Source/FullScreenPokemon.ts

```diff
@@ -2918,6 +2940,85 @@ module FullScreenPokemon {
          * @param direction   The direction border being changed.
          */
         setThingBordering(thing: IThing, other: IThing, direction: Direction): void {
+            if (thing.title === "Ledge") {
+                let borders: IThing[] = (<ILedge>thing).border[direction];
+                let placed: boolean = false;
+                // console.log("collided with ledge. setting border");
+                // console.log("ledge top/left is " + other.top + "/" + other.left);
+                for (let i: number = 0; i < borders.length; i += 1) {
+                    // the character has been removed via killNormal or the like
+                    if (!borders[i].alive) {
+                        if (placed) {
+                            borders.splice(i, 1);
+                            i--;
+                        } else {
+                            borders[i] = other;
+                            placed = !placed;
+                        }
+                    } else {
+                        /*if (borders[i] === other) {
+                            placed = !placed;
+                            continue;
+                        }*/
+                        // the character is alive so check to see if its still bordering the ledge
+                        // console.log("character top/left " + thing.top + "/" + thing.left);
+                        // if (!placed) {
+                        switch (thing.direction) {
+                            case 0:
+                                if (!placed && borders[i].top !== thing.top - 32 && borders[i].left !== other.left) {
```

Comment:
Each of these long if statements contains the same logic excluding the conditionals. Separate those into utility functions so you don't have to repeat code.